six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -195,33 +195,42 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,
         {
             if (desid == "XML_DATA_CONTENT")
             {
-                // We better have a user-defined subheader.  For now we
-                // don't actually store off the values, but we do check its
-                // tag and size.  Note that we need to check the subheader
+                // Check whether DES is SICD/SIDD, or something of a different
+                // sort. Note that we need to check the subheader
                 // length first rather than calling tre.getCurrentSize() in
                 // case there is no subheader (in which case
                 // subheader.getSubheaderFields() will throw with a
                 // NITRO-specific message).
                 const sys::Uint64_T
                         subheaderLength(subheader.getSubheaderFieldsLength());
                 if (subheaderLength !=
-                            Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
+                    Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
                 {
-                    std::ostringstream ostr;
-                    ostr << "Expected a user-defined subheader size of "
-                         << Constants::DES_USER_DEFINED_SUBHEADER_LENGTH
-                         << " but got " << subheaderLength;
-                    throw except::Exception(Ctxt(ostr.str()));
+                    continue;
                 }
 
                 nitf::TRE tre = subheader.getSubheaderFields();
+                std::string field;
+                try
```

Comment:
Looking at getDataType(nitf::Record& record) in sidd_1.1, line 95 it starts to loop through the record’s DESegments. It looks like the loop will always either return or throw in the first iteration. Is this by design?

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Wednesday, June 01, 2016 9:28 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Mention <mention@noreply.github.com>
Subject: Re: [ngageoint/six-library] Handling extra XML_DATA_CONTENT DESs when reading (#78)


In six/modules/c++/six/source/NITFReadControl.cpp<http://cp.mcafee.com/d/2DRPowd2gscCQm767CrL3zhOrKrpop73zhO--rv7cIcECzAQsCQrLnusd7dPhOYrhhphjvd7bz3VIhH3pAZxOLX2walgRKDbCRkdrFOVJQQsA7Dm1P_nUsCOCZRXBQSnXYyCCOehVqWdAklrEIYG7DR8OJMddECSjt-hsd7bOabatTztPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwVU349kITiAGM5oKfI3AnXCzAT2Ivmfb0EzapdIxOVJNZNwS2_id41FrwhQzZpgDB91Vs-h2eNb6BQQg1vo87Owhd47xYUWq8066MGaxMw2h0l4wM30id0M4Cy0e6xyxEwc4_wq88_G4E4jh0qa0c7f6yASYOrdIMR>:

>                  }

>

>                  nitf::TRE tre = subheader.getSubheaderFields();

> +                std::string field;

> +                try

Oh I see... after taking a look at the SIDD 1.1 branch's NITFReadControl::getDataType(), I'll revise this a little. Here's the problem... if you don't have your plugin path set, this'll throw which you'll catch and then you'll skip over a real DES. In that case, you want to throw an exception similar to what the branch does telling them they need to set their plugin path.

I'm not sure now why I didn't use exists() on the SIDD 1.1 branch... does this not behave well in these scenarios (TRE without this field, TRE with this field w/o plugin path set, TRE with this field with plugin path set)?

But... if you get an XML_DATA_CONTENT of length 5, even with your plugin path set, you won't have DESSHSI in there. So, need to account for that too.

Since this change is already going to make us depend on them setting their plugin path when reading a SICD, can you fold in the NITFReadControl::getDataType() from the SIDD 1.1 branch as part of this work too? Try to see if you can combine the logic for the two cleanly, at least for the DESSHSI piece.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<http://cp.mcafee.com/d/avndyhJ5xNxVCXMUQsCXCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKrtd791VRws_R-79IFLtuVtdB-_8FFIzAumKzp55mWbfaxVZicHs3jqpJATvAn3hOYyyODtUTsSjDdqymohBcCSgVv84t8_mk9VigunfAgzIiSbdDDUeuNSh_QPQXL6zAwYWw-MMyMCWwWyYgphg-UVXnk777T7fugu0ObzX0V5-VEVdMH7RzOMa8OCjr8sKrsvsodwLQzh0qmU4t8_mk9VigunfAgzIiNFtd40nS21YE4jh1UveeCy01xIayEs80Ag5h8c0M4zgc19Ew3xEoEq831fU6y2fWxa14Qg6yw31PNEFdLcCQp6gXCsNEvy>, or mute the thread<http://cp.mcafee.com/d/FZsS73hJ5xNxVCXMUQsCXCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKrtd791VRws_R-79IFLtuVtdB-_8FFIzAumKzp55mWbfaxVZicHs3jr9JATvAn3hOYyyODtUTsSjDdqymohBcCSgVv8lD9SlJDaIaoT2zp2BEO8vwwbwIg0U4m7f2yElh5gA4wmMd9Iy90Im2SL6g71vBqfyMn4gBV7g8m0Aro3q1toMrxmfH7BwkhBcCSgVsSU-UMr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq80LI43Vg8Cy3M-std4033ol5gUg18waygo1w96wo2jh073gNgQg62vMd44vR2k29Ewd5063DzhirupdP7a7tilYx1tZ>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.